### DISCUSSION TITLE
mergerfs.balance -p0

### DISCUSSION THREAD

**Discussion by simonmcnair** [trapexit/mergerfs#1322] [General]
Should this work please ?  I tried running it and I wouldn't really expect it to work as it's unlikely you could get the same amount of free space on every disk but it did come up with weird results ? (it was yesterday but I think, iirc, it seemed to hang when it was trying to figure out what to move).  I wonder if -p should have a minimum value of 1 ?

**Comment by simonmcnair** [comment]
Actually now I recall I think it just kept moving the last file from disk to disk because it couldn't make them all the same.

**Reply by trapexit** [reply]
First, what is your usecase? mergerfs.balance is not intended to be a general purpose tool. It was very explicitly created for people who used the wrong policy and didn't want to move stuff around manually to correct. Not something that was regularly run. The point of policies is that you shouldn't need out of band movement except in special cases. 

It's not about "making same". Percentage argument is a threshold. When to stop. Making it zero doesn't make sense because it is almost impossible for it to be zero. It defaults to 2.0 for that reason. And no, it can't be minimum of 1 because the value is a float and because it is a percentage it is a relative value. 1% of 1GB is a lot less than 1% of 1TB. I could refuse to take 0 but the better thing would be to make it stop after moving the same file. But these tools are really not intended for constant use and more examples to people. Hence why I never released them properly.

**Reply by Orski174** [reply]
Hi, if you could elaborate more on it's usage. Should I put it in fstab and let it run then remove it once it's balanced.

What's happening with me is that when I first set up mergerfs, all the files went to one drive then it hanged (they were all empty and mergerfs was using epmfs). So I cheated and solved it by cloning directories to all the disks, which works great for moving and copying stuff but not so much for linking them.

I was thinking about writing a script but came upon this first, seems like it would save me lots of trouble.

**Reply by simonmcnair** [reply]
It's a manual tool for one off supervised runs

**Reply by simonmcnair** [reply]
mergerfs is a union file system so you shouldn't need to clone things, just move them about.

**Reply by Orski174** [reply]
Thanks, I thought it was a flag in mergerfs, should've checked the readme first.

I was copying files from my old drives to these ones so I can add them to mergerfs after a wipe and initialization, which is why i remade all the directories, empty of course in the other drives.

**Reply by trapexit** [reply]
Why not just put your drives in the pool? Why spend all the time copying data around?

At he very least... why not just set the policy to mfs or whatever you want and rsync into the pool?

**Reply by Orski174** [reply]
They were a windows ldm array, and I didnt want to keep it as so, now that I'm on a Linux system.

And I did just that, ~~also does mfs affect read speeds?~~ Nvm, another process was running so ``$du`` was really slow.

**Reply by simonmcnair** [reply]
Mfs means minimum free space.  It is the policy used to choose where to write data.  Mergerfs should have very minimal impact on write speed, if any.

**Reply by trapexit** [reply]
The create policy has nothing to do with reads or writes.
